test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ObjectPath.java

```diff
@@ -148,4 +149,20 @@ private Object evaluate(String key, Object object, Stash stash) throws IOExcepti
 
         return list.toArray(new String[list.size()]);
     }
+
+    /**
+     * Create a new {@link XContentBuilder} from the xContent object underlying this {@link ObjectPath}.
+     * This only works for {@link ObjectPath} instances created from an xContent object, not from nested
+     * substructures. We throw an {@link UnsupportedOperationException} in those cases.
+     */
+    @SuppressWarnings("unchecked")
+    public XContentBuilder toXContentBuilder(XContent xContent) throws IOException {
+        XContentBuilder builder = XContentBuilder.builder(xContent);
+        if (this.object instanceof Map) {
+            builder.map((Map<String, Object>) this.object);
+        } else {
+            throw new UnsupportedOperationException("Only ObjectPath created from a map supported.");
```

Comment:
not that important but maybe we can also support this case, not extremely sure if it works but did you try this?

```
builder.startArray(); 
builder.value((List<?>) this.object);
builder.endArray();
```